File: app/src/main/java/com/example/movies/MoviesAdapter.java
package com.example.movies;

import android.content.Context;
import android.graphics.Movie;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

public class MoviesAdapter extends RecyclerView.Adapter<MoviesAdapter.MovieViewHolder> {

    private ArrayList<Search> movies;
    private int rowLayout;
    private Context context;


    public static class MovieViewHolder extends RecyclerView.ViewHolder {
        TextView movieTitle;
        ImageView poster;


        public MovieViewHolder(View v) {
            super(v);

            movieTitle = (TextView) v.findViewById(R.id.movieTitle);
            poster = (ImageView) v.findViewById(R.id.movieImg);
        }
    }

    public MoviesAdapter(ArrayList<Search> movies, int rowLayout, Context context) {
        this.movies = movies;
        this.rowLayout = rowLayout;
        this.context = context;
    }

    @Override
    public MoviesAdapter.MovieViewHolder onCreateViewHolder(ViewGroup parent,
                                                            int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(rowLayout, parent, false);
        return new MovieViewHolder(view);
    }


    @Override
    public void onBindViewHolder(MovieViewHolder holder, final int position) {
        holder.movieTitle.setText(movies.get(position).getTitle());
        Picasso.get().load(movies.get(position).getPoster())
                .placeholder(R.drawable.loader)
                .into(holder.poster);
    }

    @Override
    public int getItemCount() {
        return movies.size();
    }
}

File: app/src/main/java/com/example/movies/MoviesAdapter.kt
package com.example.movies

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.example.movies.MoviesAdapter.MovieViewHolder
import android.widget.TextView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.squareup.picasso.Picasso
import java.util.ArrayList

class MoviesAdapter(
    private val movies: ArrayList<Search>,
    private val rowLayout: Int,
    private val context: Context
) : RecyclerView.Adapter<MovieViewHolder>() {
    class MovieViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var movieTitle: TextView
        var poster: ImageView

        init {
            movieTitle = v.findViewById<View>(R.id.movieTitle) as TextView
            poster = v.findViewById<View>(R.id.movieImg) as ImageView
        }
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): MovieViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(rowLayout, parent, false)
        return MovieViewHolder(view)
    }

    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
        holder.movieTitle.text = movies[position].title
        Picasso.get().load(movies[position].poster)
            .placeholder(R.drawable.loader)
            .into(holder.poster)
    }

    override fun getItemCount(): Int {
        return movies.size
    }
}